nameresolution/consul/Readme.md

```diff
@@ -0,0 +1,128 @@
+# Consul Name Resolution
+
+The consul name resolution component gives the ability to register and resolve other "daprized" services registered on a consul estate. It is flexible in that it allows for complex to minimal configurations driving the behaviour on init and resolution.
+
+## How To Use
+
+### Dapr Registration
+
+If you are using the dapr sidecar to register your service to consul then you will need the following configuration:
+
+```yaml
+spec:
```

Comment:
Having tested this on Kubernetes, I think we should have the following at the top of the configuration.

```yaml
apiVersion: dapr.io/v1alpha1
kind: Configuration
metadata:
  name: appconfig
```

Likewise, if the application deployments need to refer to the configuration name per https://docs.dapr.io/operations/configuration/configuration-overview/#kubernetes-sidecar

I also think an issues needs to be created in dapr/docs to create content about name resolution and this component.

Reply:
sounds good, I didn't add it originally because I figured it was implied, in any case that's there now and I've added the docs issue